decidim-core/app/helpers/decidim/resource_helper.rb

```diff
@@ -16,8 +16,9 @@ module ResourceHelper
     # Returns nothing.
     def linked_resources_for(resource, type, link_name)
       linked_resources = resource.linked_resources(type, link_name).group_by { |linked_resource| linked_resource.class.name }
-
       safe_join(linked_resources.map do |klass, resources|
+        next unless resources.map { |r| r.component.published? }.any?
```

Comment:
Hi I think you can make this line more readable like this: `next unless resources.any? { |r| r.component.published? }`